optimade/models/structures.py

```diff
@@ -36,7 +35,8 @@
 )
 
 
-EPS = float_info.epsilon
+# Use machine epsilon for single point floating precision
```

Comment:
Weren't we using the machine epsilon before? But now not anymore? Now it's hard-coded, right?

Reply:
We were using double-precision FP (assuming it was being run on a CPU with double-prec registers, I guess?) which was fine for almost everyone, except those using more "legacy" programming languages that use single-precision by default ;)

Reply:
Maybe you can change this sentence. To me, machine suggests that it is the machine specific value of epsilon you are using. But instead you use the epsilon belonging to the standard 32-bit floating point format.
```suggestion
# Use the epsilon (The difference between 1 and the first number larger than 1) from the IEEE 754 32-bit floating point format. 
```
I am also wondering whether there may be cases where the difference is bigger than this. For example, when the numbers have been stored as a decimal string. In that case, the difference could be bigger, although we could wait with addressing this issue until it occurs.

Reply:
> Maybe you can change this sentence. To me, machine suggests that it is the machine specific value of epsilon you are using. But instead you use the epsilon belonging to the standard 32-bit floating point format.

Looking at [Wikipedia](https://en.wikipedia.org/wiki/Machine_epsilon), I think my usage is fine, but I get what you're saying. I think saying mach eps for 32-bit/single precision is fairly unambiguous.

> I am also wondering whether there may be cases where the difference is bigger than this. For example, when the numbers have been stored as a decimal string. In that case, the difference could be bigger, although we could wait with addressing this issue until it occurs.

The spec is quite specific about floating point, so I think this is as lenient as we can be really. If it has been stored as a decimal string, then hopefully that's because the numbers are nice enough to not be affected by rounding. We can see how it goes on the dashboard, as you suggest (this already fixes some problems for COD).

Reply:
Sorry, I did not realize "machine epsilon" was the whole term and not just "epsilon".